### PR TITLE
Refactor: assign settings in groups

### PR DESCRIPTION
**Pull request by kyklish**
Changes proposed in this pull request:

-   assign settings in groups (for my new settings in other PRs, that are already in group 'formatter')
-   sort settings and `enum ConfigKey` in alphabet order

Settings

Notifying @mark-wiemer

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -87,27 +87,27 @@
...
                     "default": "C:/Program Files/AutoHotkey/AutoHotkey.chm",
                     "description": "Path to the AHK Help document."
                 },
-                "ahk++.showDebugButton": {
+                "ahk++.language.enableIntellisense": {
+                    "type": "boolean",
+                    "default": true,
+                    "description": "Enable IntelliSense (Preview). Changes take effect after reload."
+                },
+                "ahk++.ui.showDebugButton": {
```
Comment (mark-wiemer) [inline package.json:110]: VS Code automatically renders `ui` as `Ui` which I think looks really gross. Let's just show `Menu` for now. See [Extension API - configuration](https://code.visualstudio.com/api/references/contribution-points#contributes.configuration). I thought there was an issue for this in VS Code, but I've lost it.

```suggestion
                "ahk++.menu.showDebugButton": {
```

File: package.json [hunk 2]
```diff
@@ -259,7 +259,7 @@
             "editor/title": [
                 {
                     "command": "ahk++.debug",
-                    "when": "editorLangId == ahk && config.ahk++.showDebugButton",
+                    "when": "editorLangId == ahk && config.ahk++.ui.showDebugButton",
```
Comment (mark-wiemer) [inline package.json:262]:
```suggestion
                    "when": "editorLangId == ahk && config.ahk++.menu.showDebugButton",
```

**Review by mark-wiemer — CHANGES_REQUESTED** [review]

**Comment by kyklish** [comment]
Do I need re-request review after I made changes?

**Review by mark-wiemer — APPROVED** [review]

**Comment by mark-wiemer** [comment]
> Do I need re-request review after I apply suggestions from code review?

No, I will actively monitor changes to PRs. This month I've been a bit busy, but I'm finally getting back to work today :)

**Pull request merged**